README.md

```diff
@@ -0,0 +1,53 @@
+# P2PMEM PCIe Linux Driver
+
+## Disclimer
+
+This driver exposes p2pmem to userspace without taking appropriate
+safety measures to ensure it is used correctly. If you do certain
+things with the pointers obtained by mmapping the /dev/p2pmemX exposed
+by this driver bad things will probably happen. Consider yourself
+warned.
+
+## Introduction
+
+This is a standlone PCIe driver that ties into the [p2pmem][1]
+framework. It can be used for any PCIe end-point that exposes one or
+more PCIe BAR(s) that can be used for DMAs and whose PCIe driver has
+been updated to add one or more p2pdma resources and register said
+resource(s) against the p2pdma framework.
```

Comment:
> It can be used for any PCIe end-point that exposes one or
more PCIe BAR(s) that can be used for DMAs and whose PCIe driver has
been updated to add one or more p2pdma resources and register said
resource(s) against the p2pdma framework.

This is a very awkward sentence with some mixed tensing. Maybe something like "It can be used for any PCIe end-point devices that have registered one or more exposed PCIe BAR(s) with the p2pdma framework."

Maybe even add in an "i.e. NVMe CMBs" at the end here.

Reply:
Thanks. I used your sentence but dropped "exposed" because its redundent (a BAR is alwats exposed by definition).